apps/minifront/src/components/swap/swap-form/simulate-swap.tsx

```diff
@@ -3,16 +3,33 @@ import { SimulateSwapResult } from './simulate-swap-result';
 import { AllSlices } from '../../../state';
 import { useStoreShallow } from '../../../utils/use-store-shallow';
 import { EstimateButton } from './estimate-button';
+import { useEffect } from 'react';
 
 const simulateSwapSelector = (state: AllSlices) => ({
   simulateSwap: state.swap.instantSwap.simulateSwap,
+  resetSimulateSwap: state.swap.instantSwap.reset,
   disabled:
     state.swap.txInProgress || !state.swap.amount || state.swap.instantSwap.simulateSwapLoading,
   result: state.swap.instantSwap.simulateSwapResult,
+  amount: state.swap.amount,
 });
 
+// Automatically trigger swap simulation on user input
 export const SimulateSwap = ({ layoutId }: { layoutId: string }) => {
-  const { simulateSwap, disabled, result } = useStoreShallow(simulateSwapSelector);
+  const { simulateSwap, resetSimulateSwap, disabled, result, amount } =
+    useStoreShallow(simulateSwapSelector);
+
+  useEffect(() => {
+    if (!disabled && amount && parseFloat(amount) > 0) {
+      // Prevents re-triggering the swap simulation if it's already computed
+      if (!result) {
+        void simulateSwap();
```

Comment:
comment: we can also wrap this in a framer motion div for smoother animations

Reply:
Could be future work